test/shared_pubspec_tests.dart

```diff
@@ -0,0 +1,294 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:meta/meta.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:test/test.dart';
+import 'package:yaml/yaml.dart';
+
+import 'util.dart';
+
+/// Throws if [yaml] is invalid.
```

Comment:
Same comment (directed at myself) as below:
```suggestion
/// Throws if [yaml] is an invalid pubspec, either due to:
///
/// - being unparseable 
/// - having incorrect structure (this check is not comprehensive)
```